rust/src/action.rs

```diff
@@ -845,32 +844,75 @@ impl Action {
 
 /// Operation performed when creating a new log entry with one or more actions.
 /// This is a key element of the `CommitInfo` action.
-#[derive(Serialize, Deserialize, Debug)]
+#[allow(clippy::large_enum_variant)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
 pub enum DeltaOperation {
+    /// Represents a Delta `Create` operation.
+    /// Would usually only create the table, if also data is written,
+    /// a `Write` operations is more appropriate
+    Create {
+        /// The save mode used during the create.
+        mode: SaveMode,
+        /// The storage location of the new table
+        location: String,
+        /// The min reader and writer protocol versions of the table
+        protocol: Protocol,
+        /// Metadata associated with the new table
+        metadata: DeltaTableMetaData,
+    },
+
     /// Represents a Delta `Write` operation.
     /// Write operations will typically only include `Add` actions.
+    #[serde(rename_all = "camelCase")]
     Write {
         /// The save mode used during the write.
         mode: SaveMode,
         /// The columns the write is partitioned by.
-        partitionBy: Option<Vec<String>>,
+        partition_by: Option<Vec<String>>,
         /// The predicate used during the write.
         predicate: Option<String>,
     },
     /// Represents a Delta `StreamingUpdate` operation.
+    #[serde(rename_all = "camelCase")]
     StreamingUpdate {
         /// The output mode the streaming writer is using.
-        outputMode: OutputMode,
+        output_mode: OutputMode,
         /// The query id of the streaming writer.
-        queryId: String,
+        query_id: String,
         /// The epoch id of the written micro-batch.
-        epochId: i64,
+        epoch_id: i64,
     },
     // TODO: Add more operations
 }
 
+impl DeltaOperation {
+    /// Retrieve basic commit information to be added to Delta commits
+    pub fn get_commit_info(&self) -> Map<String, Value> {
+        let mut commit_info = Map::<String, Value>::new();
+        let operation = match &self {
+            DeltaOperation::Create { .. } => "delta-rs.Create",
+            DeltaOperation::Write { .. } => "delta-rs.Write",
+            DeltaOperation::StreamingUpdate { .. } => "delta-rs.StreamingUpdate",
+        };
+        commit_info.insert(
+            "operation".to_string(),
+            serde_json::Value::String(operation.into()),
+        );
+
+        if let Ok(serde_json::Value::Object(map)) = serde_json::to_value(self) {
+            commit_info.insert(
+                "operationParameters".to_string(),
+                map.values().next().unwrap().clone(),
```

Comment:
just checked again, turns out we are adding all the parameter keys and values. The map form `serde_json::to_value(self)` looks something like `{"create": {"key": "value", ...}}` so I think we can assume that we always have exactly one relevant key and are adding the full "inner" map.

Reply:
Oh i see, we are trying to get rid of the enum variant layer, thanks for the clarification :+1:  Could you a comment in the code to mention what you said in the comment? It was obvious to me when I first read it.

Ideally, it's better to manage each enum option as its own struct, then generate the option json object from within the first `match &self {` block. But since this is not a hot code path, I think it's fine to leave this as is for now :)